plugins/download_statistics/server/__init__.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+
+from girder import events
+from girder.constants import AccessType
+from girder.utility.model_importer import ModelImporter
+
+
+def downloadItemStartEvent(event):
+    # Ony count download as started if offset is 0
+    itemModel = ModelImporter.model('item')
+    item = itemModel.load(event.info['id'], force=True)
+    firstRequest = 'offset' not in event.info['params'] or event.info['params']['offset'] == '0'
+
+    # Get all files in item to increment their download count
+    files = itemModel.childFiles(item=item)
+    for file in files:
```

Comment:
Rather than do `n` different increment operations for every file in this item, we could be more efficient by pulling the query logic out of `childFiles`, and using it to increment all the matched files with a single database request.